src/js/thirdparty/ws.js

```diff
@@ -50,49 +50,81 @@ class BunWebSocket extends EventEmitter {
     // TODO: options
   }
 
-  on(event, listener) {
+  #onOrOnce(event, listener, once) {
     if (event === "unexpected-response" || event === "upgrade" || event === "redirect") {
       emitWarning(event, "ws.WebSocket '" + event + "' event is not implemented in bun");
     }
     const mask = 1 << eventIds[event];
     if (mask && (this.#eventId & mask) !== mask) {
       this.#eventId |= mask;
       if (event === "open") {
-        this.#ws.addEventListener("open", () => {
-          this.emit("open");
-        });
+        this.#ws.addEventListener(
+          "open",
+          () => {
+            this.emit("open");
+          },
+          once,
+        );
       } else if (event === "close") {
-        this.#ws.addEventListener("close", ({ code, reason, wasClean }) => {
-          this.emit("close", code, reason, wasClean);
-        });
+        this.#ws.addEventListener(
+          "close",
+          ({ code, reason, wasClean }) => {
+            this.emit("close", code, reason, wasClean);
+          },
+          once,
+        );
       } else if (event === "message") {
-        this.#ws.addEventListener("message", ({ data }) => {
-          const isBinary = typeof data !== "string";
-          if (isBinary) {
-            this.emit("message", this.#fragments ? [data] : data, isBinary);
-          } else {
-            let encoded = encoder.encode(data);
-            if (this.#binaryType !== "arraybuffer") {
-              encoded = Buffer.from(encoded.buffer, encoded.byteOffset, encoded.byteLength);
+        this.#ws.addEventListener(
+          "message",
+          ({ data }) => {
+            const isBinary = typeof data !== "string";
+            if (isBinary) {
+              this.emit("message", this.#fragments ? [data] : data, isBinary);
+            } else {
+              let encoded = encoder.encode(data);
+              if (this.#binaryType !== "arraybuffer") {
+                encoded = Buffer.from(encoded.buffer, encoded.byteOffset, encoded.byteLength);
```

Comment:
If we have access to builtins here then:
```suggestion
                encoded = $Buffer.$from(encoded.buffer, encoded.byteOffset, encoded.byteLength);
```